addons/web/controllers/main.py

```diff
@@ -1025,7 +1027,7 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas
                       filename_field='datas_fname', unique=None, filename=None, mimetype=None,
                       download=None, width=0, height=0, crop=False, access_token=None, avoid_if_small=False,
                       upper_limit=False, placeholder='placeholder.png', **kw):
-        status, headers, content = request.env['ir.http'].binary_content(
+        status, headers, content = request.env['ir.http'].with_context(allowed_company_ids=','.join(str(elem) for elem in request.env.user.company_ids.ids)).binary_content(
```

Comment:
should not be needed, see https://github.com/odoo/odoo/pull/32341/commits/2cc360b61889e4c2831c815e1f4358bcd75ccff3#diff-2c418cb78b7c562a79e7032645a537d2R876